File: frontend/src/types/topic.ts
export interface Topic {
  id: number;
  name: string;
  slug: string;
  description?: string;
  created_at: string;
}

export interface TopicFollow {
  id: number;
  user_id: number;
  topic_id: number;
  created_at: string;
  topic?: Topic;
}

File: backend/models/topic.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Topic struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	Name        string         `gorm:"unique;not null" json:"name"`
	Slug        string         `gorm:"unique;not null;index" json:"slug"`
	Description string         `gorm:"type:text" json:"description"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}
